src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/SALauncher.java

```diff
@@ -126,7 +126,9 @@ private static boolean jmapHelp() {
         System.out.println("    <no option>             To print same info as Solaris pmap.");
         System.out.println("    --heap                  To print java heap summary.");
         System.out.println("    --binaryheap            To dump java heap in hprof binary format.");
-        System.out.println("    --dumpfile <name>       The name of the dump file.");
+        System.out.println("        dump options:");
+        System.out.println("          --dumpfile <name> The name of the dump file.");
+        System.out.println("          --gz <1-9>        The compression level for gzipped dump file.");
```

Comment:
This is a little awkward, and will get worse if later you add an option to dump in GXL format (format=x). Maybe for `--dumpfile` and `--gz` just add a comment "Only valid with --binaryheap".